docs/index.stories.js

```diff
@@ -0,0 +1,246 @@
+import LinkTo from '@storybook/addon-links/react';
+import { setOptions } from '@storybook/addon-options';
+import { exampleStory } from '../.storybook/lucid-docs-addon';
+import readmeText from '!!raw-loader!../README.md';
+import introText from '!!raw-loader!./intro.md';
+import childComponentsText from '!!raw-loader!./child-components.md';
+import hybridComponentsText from '!!raw-loader!./hybrid-components.md';
+import computedPropsText from '!!raw-loader!./computed-props.md';
+import LucidLogo from './LucidLogo';
+import SyntaxHighlighter, {
+	registerLanguage,
+} from 'react-syntax-highlighter/prism-light';
+import jsx from 'react-syntax-highlighter/languages/prism/jsx';
+import okaidia from 'react-syntax-highlighter/styles/prism/okaidia';
+import '../src/index.less';
+
+registerLanguage('jsx', jsx);
+
+const loadAllKeys = (reqContext, rawContext) => {
+	return _.map(reqContext.keys(), key => ({
+		key,
+		module: reqContext(key),
+		raw: rawContext(key),
+	}));
+};
+
+const getDefaultExport = module => {
+	if (module.__esModule) {
+		return module.default;
+	}
+	return module;
+};
+
+const getExamplesFromContext = (reqExamples, rawContext) =>
+	_.map(loadAllKeys(reqExamples, rawContext), ({ key, module, raw }) => ({
+		name: _.join(_.reject(_.words(key), w => /^(\d+)|jsx?$/.test(w)), ' '),
+		Example: getDefaultExport(module),
+		source: raw,
+	}));
+
+const checkIconSVG = `<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE svg>
+<svg xmlns="http://www.w3.org/2000/svg" width="14" height="14" viewBox="0 0 16 16" preserveAspectRatio="xMidYMid meet" style="fill: rgb(42, 187, 176)">
+	<path d="M11.92 5.19c.285.284.285.748 0 1.032l-4.932 4.98c-.285.286-.748.286-1.033 0L3.47 8.85c-.286-.287-.286-.748 0-1.034l.287-.343c.285-.286.747-.286 1.033 0l1.747 1.614 4.13-4.127c.284-.285.748-.285 1.033 0l.22.23z"></path>
+</svg>
+`;
+
+const styles = {
+	link: {
+		//color: '#3f6486',
+		color: '#2abbb0',
+		textDecoration: 'underline',
+		cursor: 'pointer',
+		outline: 'none',
+	},
+	ul: {
+		listStyleImage: `url('data:image/svg+xml;base64,${window.btoa(
+			checkIconSVG
+		)}')`,
+	},
+	li: {
+		margin: '8px 0',
+	},
+};
+
+const compile = marksy({
+	createElement: React.createElement,
+	highlight: (language, code) =>
+		ReactDOMServer.renderToStaticMarkup(
+			<SyntaxHighlighter language={language || 'jsx'} style={okaidia}>
+				{code}
+			</SyntaxHighlighter>
+		),
+	elements: {
+		a: props => <a {...props} style={styles.link} />,
+		ul: props => <ul {...props} style={styles.ul} />,
+		li: props => <li {...props} style={styles.li} />,
+	},
+	components: {
+		LinkTo: props => <LinkTo {...props} style={styles.link} />,
+	},
+});
+
+class ArticlePage extends React.Component {
+	constructor(...args) {
+		super(...args);
+	}
+
+	componentDidMount() {
+		if (typeof window !== 'undefined') {
+			window.document.documentElement.scrollTop = 0;
+		}
+	}
+
+	render() {
+		const { children } = this.props;
+		setOptions({
```

Comment:
does this have to be called every render? or could it be moved out of render?

Reply:
it could be moved to `componentDidMount`